pkg/sql/logictest/testdata/logic_test/alter_type

```diff
@@ -80,3 +80,23 @@ query T
 SELECT ARRAY['hi']::___why
 ----
 {hi}
+
+statement ok
```

Comment:
ensure that you can't create the old value -- ie `SELECT 'jim'::names` after the rename should fail